README.md

```diff
@@ -1,7 +1,83 @@
-# CFL with AssetSwapper
+![alt text](/banner.png "Get started with 0x API")
 
+# Get started with 0x API
+
+This is a repo containing toy examples of filling 0x-API quotes either directly with web3 or through a smart contract.
+
+## Installation
+Clone this repo then, from inside the project, run:
+```bash
+yarn -D
+# or
+npm install --dev
+```
+
+This will also compile the contracts in `contracts/` and produce artifacts in `build/`.
+
+## Examples
+The following example scripts are included:
+
+| Script | Guide | Description |
+|--------|-------|-------------|
+| `src/direct-swap.js` |  [Swap tokens with 0x API](https://0x.org/docs/guides/swap-tokens-with-0x-api) | Perform a token swap with web3. |
+| `src/swap-contract.js` | [Use 0x API liquidity in your smart contracts](https://0x.org/docs/guides/use-0x-api-liquidity-in-your-smart-contracts) | Perform a token swap in a smart contract. |
+
+### Running the examples locally (forked mainnet)
+The examples can be run locally (without actually mining transactions) through the magic of ganache forking. You will first need to start a forked ganache instance with the following command, replacing `ETHEREUM_RPC_URL` with the HTTP or websocket RPC URL of your mainnet ethereum node (e.g., Infura mainnet):
+
+```bash
+RPC_URL=ETHEREUM_RPC_URL npm run start-fork
+```
+
+#### Direct swap
+To run the direct swap example, in a separate terminal run:
+```bash
+npm run swap-fork
 ```
-yarn
-yarn build
-yarn test
+
+#### Contract swap
+To run the contract swap example, in a seperate terminal run:
+```bash
+npm run deploy-fork # Only need to do this once per ganache instance.
+npm run swap-contract-fork
 ```
+
+### Running the examples on mainnet
+You can also run the examples and perform actual swaps that will get mined with a little effort:
+
+1. Modify the `mnemonic` in `package.json` to one only you know.
```

Comment:
Might be worth to emphasize that you should **not**  under any circumstance use the ganache mnemonic 😅